frontend/lint/eslint-rules/no-unconditional-metabase-links-render.js

```diff
@@ -0,0 +1,214 @@
+            named: "getShowMetabaseLinks",
+            source: "metabase/selectors/whitelabel",
+          })
+        ) {
+          isGetShowMetabaseLinksSelectorImported = true;
+        }
+      },
+      CallExpression(node) {
+        // call `getDocsUrl` selector
+        if (
+          isGetDocsUrlSelectorImported &&
+          !isGetShowMetabaseLinksSelectorImported &&
+          node?.callee?.type === "Identifier" &&
+          node?.callee?.name === "getDocsUrl"
+        ) {
+          context.report({
+            node,
+            message: ERROR_MESSAGE,
+          });
+        }
+
+        // call `getLearnUrl` selector
+        if (
+          isGetLearnUrlSelectorImported &&
+          !isGetShowMetabaseLinksSelectorImported &&
+          node?.callee?.type === "Identifier" &&
+          node?.callee?.name === "getLearnUrl"
+        ) {
+          context.report({
+            node,
+            message: ERROR_MESSAGE,
+          });
+        }
+
+        // call `MetabaseSettings.learnUrl` or `MetabaseSettings.docsUrl`
+        if (
+          isMetabaseSettingsImported &&
+          !isGetShowMetabaseLinksSelectorImported &&
+          node?.callee?.object?.type === "Identifier" &&
+          node?.callee?.object?.name === "MetabaseSettings" &&
+          ["learnUrl", "docsUrl"].includes(node?.callee?.property?.name)
+        ) {
+          context.report({
+            node,
+            message: ERROR_MESSAGE,
+          });
+        }
+      },
+      Literal(node) {
+        if (typeof node.value !== "string") {
+          return;
+        }
+
+        if (
+          LITERAL_METABASE_URL_REGEX.exec(node.value) &&
+          !isGetShowMetabaseLinksSelectorImported
+        ) {
+          context.report({
+            node,
+            message: ERROR_MESSAGE,
+          });
+        }
+      },
+      TemplateLiteral(node) {
+        const quasis = node.quasis;
+        quasis.forEach(quasi => {
+          if (LITERAL_METABASE_URL_REGEX.exec(quasi.value.raw)) {
+            context.report({
+              node,
+              message: ERROR_MESSAGE,
+            });
+          }
+        });
+      },
+      Program() {
+        const comments = context.getSourceCode().getAllComments();
+
+        const ESLINT_DISABLE_BLOCK_REGEX =
+          /eslint-disable\s+no-unconditional-metabase-links-render/;
+        const ESLINT_DISABLE_LINE_REGEX =
+          /eslint-disable-next-line\s+no-unconditional-metabase-links-render/;
+        const ALLOWED_ESLINT_DISABLE_LINE_REGEX =
+          /eslint-disable-next-line\s+no-unconditional-metabase-links-render -- \w+/;
+        comments.forEach(comment => {
+          if (ESLINT_DISABLE_BLOCK_REGEX.exec(comment.value)) {
+            const { start, end } = comment.loc;
+            context.report({
+              loc: {
+                start: { line: start.line - 1, column: start.column },
+                end: { line: end.line - 1, column: end.column },
```

Comment:
It seems that disabling using block comment will prevent the error on the actual comment line. That makes sense because the rule is disabled _starting from_ the actual comment line.

So the hack would be to display the error on the line before. This works even if the block comment is the first line of the file.

The line comment works because it only disables the rule in the next line as the comment suggested `eslint-disable-next-line`